sakura_core/charset/charcode.h

```diff
@@ -126,50 +113,50 @@ namespace WCODE
 	bool IsControlCode(wchar_t wc);
 
 	//!改行文字であるかどうか
-	inline bool IsLineDelimiter(wchar_t wc, bool ext)
+	[[nodiscard]] inline bool IsLineDelimiter(wchar_t wc, bool ext)
 	{
 		return wc==CR || wc==LF || (ext && (wc==0x85 || wc==0x2028 || wc==0x2029));
 	}
-	inline bool IsLineDelimiterBasic(wchar_t wc)
+	[[nodiscard]] inline bool IsLineDelimiterBasic(wchar_t wc)
 	{
 		return wc==CR || wc==LF;
 	}
-	inline bool IsLineDelimiterExt(wchar_t wc)
+	[[nodiscard]] inline bool IsLineDelimiterExt(wchar_t wc)
 	{
 		return wc==CR || wc==LF || wc==0x85 || wc==0x2028 || wc==0x2029;
 	}
 
 	//!単語の区切り文字であるかどうか
-	inline bool IsWordDelimiter(wchar_t wc)
+	[[nodiscard]] inline bool IsWordDelimiter(wchar_t wc)
 	{
 		return wc==SPACE || wc==TAB || IsZenkakuSpace(wc);
 	}
 
 	//!インデント構成要素であるかどうか。bAcceptZenSpace: 全角スペースを含めるかどうか
-	inline bool IsIndentChar(wchar_t wc,bool bAcceptZenSpace)
+	[[nodiscard]] inline bool IsIndentChar(wchar_t wc,bool bAcceptZenSpace)
 	{
 		if(wc==TAB || wc==SPACE)return true;
 		if(bAcceptZenSpace && IsZenkakuSpace(wc))return true;
 		return false;
 	}
 
 	//!空白かどうか
-	inline bool IsBlank(wchar_t wc)
+	[[nodiscard]] inline bool IsBlank(wchar_t wc)
 	{
 		return wc==TAB || wc==SPACE || IsZenkakuSpace(wc);
 	}
 
 	//!ファイル名に使える文字であるかどうか
 	inline bool IsValidFilenameChar(const wchar_t wc)
 	{
-		static const wchar_t* table = L"<>?\"|*";
+		constexpr wchar_t table[] = L"<>?\"|*";
 
-		if(wcschr(table,wc)!=NULL)return false; //table内の文字が含まれていたら、ダメ。
-		else return true;
+		//table内の文字が含まれていたら、ダメ。
+		return wcschr(table, wc) == nullptr;
```

Comment:
指摘というより呟きっす。
```suggestion
		return !::wcschr(table, wc);
```
こう書いても同じ意味です。どっちがいい！ということではありません・・・。